File: src/main/java/com/kingsoft/thrift/PersonServiceImpl.java
package com.kingsoft.thrift;

import org.apache.thrift.TException;
import thrift.generated.DataException;
import thrift.generated.Person;
import thrift.generated.PersonService;

/**
 * @ClassName PersonServiceImpl
 * @Description TODO
 * @Author 黄育佳
 * @Date 2020/4/6 0:56
 * @Version 1.0
 **/
public class PersonServiceImpl implements PersonService.Iface{
    @Override
    public Person getPersonByUsername(String username) throws DataException, TException {
        System.out.println("getPersonByUsername : Get Client Param:" + username);
        Person person = new Person();
        person.setUsername(username);
        person.setAge(20);
        person.setMarried(false);
        return person;
    }

    @Override
    public void savePerson(Person person) throws DataException, TException {
        System.out.println("savePerson....");
        System.out.println(person.getUsername());
        System.out.println(person.getAge());
        System.out.println(person.isMarried());
    }
}

File: pythrift/PersonServiceImpl.py
# coding=utf-8

from py.thrift.generated import ttypes


class PersonServiceImpl:

    def getPersonByUsername(self, username):
        print "Get client param:" + username
        person = ttypes.Person()
        person.username = username
        person.age = 12
        person.married = False
        return person

    def savePerson(self, person):
        print "Get client param:"
        print person.username
        print person.age
        print person.married
